src/PROPOSAL/detail/PROPOSAL/propagation_utility/PropagationUtilityInterpolant.cxx

```diff
@@ -94,8 +96,18 @@ double UtilityInterpolant::GetUpperLimit(double upper_limit, double rnd)
     initial_guess.x = NAN;
     initial_guess.upper = upper_limit;
 
-    return cubic_splines::find_parameter(
-        *interpolant_, integrated_to_upper - rnd, initial_guess);
+    try {
+        return cubic_splines::find_parameter(
+                *interpolant_, integrated_to_upper - rnd, initial_guess);
+    } catch (...) {
```

Comment:
This shouldn't be activated for any exception.

Reply:
Is it possible to catch a specific boost exception in PROPOSAL if PROPOSAL does not directly depend on boost?
I agree with your point, but I am not sure if this is possible.